moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -76,17 +77,61 @@ moveit::core::RobotStatePtr CurrentStateMonitor::getCurrentState() const
   return moveit::core::RobotStatePtr(result);
 }
 
-ros::Time CurrentStateMonitor::getCurrentStateTime() const
+ros::Time CurrentStateMonitor::getCurrentStateTimeHelper(const std::string& group) const
+{
+  const std::vector<const moveit::core::JointModel*>* active_joints = &robot_model_->getActiveJointModels();
+  if (!group.empty())
+  {
+    const moveit::core::JointModelGroup* jmg = robot_model_->getJointModelGroup(group);
+    if (jmg)
+    {
+      active_joints = &jmg->getActiveJointModels();
+    }
+    else
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "There is no group with the name " << std::quoted(group) << "!");
+      return ros::Time(0.0);
+    }
+  }
+  std::optional<ros::Time> oldest_state_time = std::nullopt;
```

Comment:
Instead of `std::optional` you could directly use `ros::Time()` as the null time:
```suggestion
  auto oldest_state_time = ros::Time();
```

Reply:
Sure, you still need to explicitly check for the special case here: `if (!oldest_state_time.isZero())`

Reply:
That would work yes 👍🏼 This still contradicts the extremely edge case of having a valid joint state with 0.0 timestamp though 😄 Admittedly, it is not realistic.

Reply:
Done.